onnx/test/optimizer_test.py

```diff
@@ -1120,6 +1120,38 @@ def test_eliminate_nop_dropout(self):  # type: () -> None
                 assert len(optimized_model.graph.node) == 1
                 assert optimized_model.graph.node[0].op_type == "Log"
 
+    def test_fuse_reduction_unsqueeze(self):  # type: () -> None
+        for reduction in ["ReduceL1", "ReduceL2", "ReduceLogSum",
+                          "ReduceLogSumExp", "ReduceMax", "ReduceMean",
+                          "ReduceMin", "ReduceProd", "ReduceSum", "ReduceSumSquare"]:
+            for axes1 in [[1], [1, 2]]:
+                for axes2 in [[1], [1, 2]]:
+                    for keepdim in [False, True]:
+                        input_shape = (5, 7, 9)
+                        output_shape = input_shape  # type: Tuple[int, ...]
+                        if not keepdim and axes1 == axes2:
+                            output_shape = tuple([x for i, x in enumerate(output_shape) if i not in axes1])
```

Comment:
output_shape seems still incorrect.
For example:
axes1 = [1]
axes2 = [1]
and keepdim = false
outshape should be (5, 1, 9)

Reply:
fixed. I missed a ton of cases. Sorry about that.